### PR TITLE
Fix V-DOGE-VUL-007

### PR DESCRIPTION
**Pull request by DarianShawn**
# Description

The PR fixes nil reference attack in transaction gossip request.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: txpool/txpool_test.go [hunk 1]
```diff
@@ -388,6 +388,27 @@ func TestDropKnownGossipTx(t *testing.T) {
...
+	pool.SetSigner(&mockSigner{})
+
+	assert.NotPanics(t, func() {
+		// type assertion
+		pool.addGossipTx(&types.Block{Header: &types.Header{Number: 10}})
+	})
+
+	assert.Nil(t, pool.accounts.get(addr1), "addr in txpool should be nil")
+
+	mightPanicTx := newTx(addr1, 1, 1)
```
Comment (abrahamcruise321) [inline txpool/txpool_test.go]: should be *proto.Txn.

**Review by abrahamcruise321 — APPROVED** [review]
ok

File: txpool/txpool_test.go (hunk 1 above)
Comment (DarianShawn) [reply txpool/txpool_test.go]: Updated, review it please.

Comment (DarianShawn) [reply txpool/txpool_test.go]: mark it resolved, which blocks the merging

**Pull request merged**